Common/Systems/AltUseUISystem.cs

```diff
@@ -67,11 +67,11 @@ private void DrawChargeBar()
 			return;
 		}
 
-		Vector2 center = player.Center - Main.screenPosition + Vector2.UnitY * player.height;
+		Vector2 center = new Vector2(Main.screenWidth, Main.screenHeight) / 2f + Vector2.UnitY * player.height;
```

Comment:
The reason I had programmed it to follow the player specifically is because the player isn't always in the center of the screen: 
![image](https://github.com/user-attachments/assets/f62f7223-5c96-4b41-9f3d-2628807d8698)

How was it incorrect in the first place?

Reply:
Here is where I saw it was changed:
https://github.com/Path-of-Terraria/PathOfTerraria/commit/d2a25535bd396cb4c41e169ce79d7d4227a75610#diff-496c6c74034cff7e52eeac6df2ab8351b3478f0415a47ed5b171c3a557eaceb5R70

I only reverted this change and noticed it started to work.

That said, I forgot about the camera moving on the edges of the map. I have a screenshot of what it looked like for me prior to this change. I changed all resolutions and noticed it was always not in the correct place.

Reply:
Looks like the commit you linked was a merge of my changes to tomat's branch. I don't recall it working poorly, but either way we'll need to change this. Especially given a good amount of mods change the camera position, as do niche vanilla items like the Sniper Rifle.

Reply:
Apologies for the delay, confirmed this fixes it. Updated PR